src/config/macro_names.rs

```diff
@@ -0,0 +1,81 @@
+//! This module contains types and functions to support formatting specific macros.
+
+use itertools::Itertools;
+use std::{fmt, str};
+
+use serde::{Deserialize, Serialize};
+use serde_json as json;
+use thiserror::Error;
+
+/// Defines the name of an input - either a file or stdin.
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Ord, PartialOrd, Deserialize, Serialize)]
+pub struct MacroName(String);
+
+impl fmt::Display for MacroName {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl From<MacroName> for String {
+    fn from(other: MacroName) -> Self {
+        other.0
+    }
+}
+
+/// A set of macro names.
+#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
+pub struct MacroNames(Vec<MacroName>);
+
+impl fmt::Display for MacroNames {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0.iter().format(", "))
+    }
+}
+
+impl MacroNames {
+    /// Return the underlying macro names, as an iterator of strings.
+    pub(crate) fn into_name_strings(self) -> impl Iterator<Item = String> {
+        self.0.into_iter().map(Into::into)
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum MacroNamesError {
+    #[error("{0}")]
+    Json(json::Error),
+}
+
+// This impl is needed for `Config::override_value` to work for use in tests.
+impl str::FromStr for MacroNames {
+    type Err = MacroNamesError;
+
+    fn from_str(s: &str) -> Result<MacroNames, Self::Err> {
+        json::from_str(s).map_err(MacroNamesError::Json)
+    }
+}
```

Comment:
I believe the current `IgnoreList` for ignored files always raises an error in it's `FromStr` impl to disallow setting an ignore list via the command line using `--config`.

I don't feel strongly either way, but I wonder if we should also require that the `skip_macro_names` list only be provided via the `rustfmt.toml`?

Reply:
Pragmatically, I use this implementation in the `tests` to pass in the option for test cases - there is prior art for this with `--file-lines`.

I don't see particularly why you'd want to forbid setting this from the command line?